SurfaceTopography/IO/NC.py

```diff
@@ -129,44 +136,91 @@ def __init__(self, fobj, communicator=None):
+
+        # Determine physical sizes of topography
+        if self._n_dim is not None:
+            # This is a nonuniform topography
+            pass
+        else:
+            # This is a uniform topography...
             if self._y_var is not None:
-                self._physical_sizes = (self._x_var.length, self._y_var.length)
-            try:
-                self._periodic = self._x_var.periodic != 0
-            except AttributeError:
-                pass
-            try:
-                self._info['unit'] = mangle_height_unit(self._x_var.length_unit)
-            except AttributeError:
-                pass
+                # ...and it is 2D
+                try:
+                    self._physical_sizes = (self._x_var.length, self._y_var.length)
+                except AttributeError:
+                    pass
+            else:
+                # ...and it is 1D (a line scan)
+                try:
+                    self._physical_sizes = (self._x_var.length)
+                except AttributeError:
+                    pass
+
+        # Determine if the topography is periodic
+        try:
+            self._periodic = self._x_var.periodic != 0
+        except AttributeError:
+            pass
+
+        # Determine unit of topography
+        try:
+            self._info['unit'] = mangle_length_unit_utf8(self._x_var.unit)
+        except AttributeError:
+            pass
 
     def __del__(self):
         self.close()
 
     def close(self):
         if self._nc is not None:
+            if self._x_var is not None:
+                del self._x_var
+            if self._y_var is not None:
+                del self._y_var
+            if self._heights_var is not None:
+                del self._heights_var
             self._nc.close()
             self._nc = None
 
     @property
     def channels(self):
-        try:
-            # netCDF4
-            nx = self._x_dim.size
-        except AttributeError:
-            # scipy.io.netcdf_file
-            nx = self._x_dim
-        try:
-            # netCDF4
-            ny = self._y_dim.size
-        except AttributeError:
-            # scipy.io.netcdf_file
-            ny = self._y_dim
+        if self._x_dim is not None:
+            # This is a uniform topography
+            try:
+                # netCDF4
+                nx = self._x_dim.size
+            except AttributeError:
+                # scipy.io.netcdf_file
+                nx = self._x_dim
+            nb_grid_pts = (nx,)
+            if self._y_dim is not None:
+                # This is a 2D topography
+                try:
+                    # netCDF4
+                    ny = self._y_dim.size
+                except AttributeError:
+                    # scipy.io.netcdf_file
+                    ny = self._y_dim
+                nb_grid_pts = (nx, ny)
+        else:
+            # This is a nonuniform line scan
+            try:
+                # netCDF4
+                nx = self._x_dim.size
```

Comment:
Does _x_dim exist in the nonuniform scans ? Or what is the purpose of _n_dim ?

Reply:
No, nonuniform scans have only `_n_dim`. The idea is to distinguish between equally spaces sets `_x_dim` and nonequally spaces sets `_n_dim`, where `n` is the number of data points. I could be convinced to just call it `x_dim` in both cases.

Reply:
But how do you detect uniform versus nonuniform ?

Reply:
From the presence of the `length` attribute, but I guess that having a different name for the dimension is clearer.

Reply:
That said, I need to check support for periodic nonuniform line scans which should actually need the length - but this seems to be broken in the code. We should probably simply not support periodic nonuniform line scans.

Reply:
See #78